app/src/main/java/org/mozilla/focus/settings/SettingsFragment.java

```diff
@@ -81,15 +90,35 @@ public boolean onPreferenceTreeClick(PreferenceScreen preferenceScreen, Preferen
             final Intent intent = InfoActivity.getPrivacyNoticeIntent(getActivity());
             startActivity(intent);
         } else if (preference.getKey().equals(resources.getString(R.string.pref_key_search_engine))) {
-            final Intent intent = new Intent(getActivity(), SettingsActivity.class);
-            intent.putExtra(FRAGMENT_RESID_INTENT_EXTRA, R.xml.search_engine_settings);
-            intent.putExtra(TITLE_RESID_INTENT_EXTRA, R.string.preference_search_installed_search_engines);
+            final Intent intent = getSettingsIntent(getActivity(),
+                    AppConstants.FLAG_MANUAL_SEARCH_ENGINE ? R.xml.search_engine_settings_featureflag_manual : R.xml.search_engine_settings,
+                    R.string.preference_search_installed_search_engines);
+            startActivity(intent);
+        } else if (preference.getKey().equals(resources.getString(R.string.pref_key_manual_add_search_engine))) {
+            final Intent intent = getSettingsIntent(getActivity(), ManualAddSearchEngineSettingsFragment.FRAGMENT_CLASS_TYPE,
+                    R.xml.manual_add_search_engine,
+                    R.string.tutorial_search_title);
             startActivity(intent);
```

Comment:
So this starts another instance of the settings activity on top with ManualAddSearchEngineSettingsFragment? Is this simpler here then just adding the fragment on top (like we do in MainActivity)? I guess one annoying thing here is the toolbar that needs to change. In MainActivity the toolbar is part of the fragments themselves - I wonder if this would work here too? Anyhow, I won't block landing this - but this part felt a bit like it is hard to understand while reading the code.

Reply:
Mmm good call - I initially did this because I ran into some problems with the support action bar, but now that I'm more familiar with it I switched to using the FragmentManager to handle these, which is a lot better.